File: Leetcode/921. Minimum Add to Make Parentheses Valid/solution.cpp
class Solution
{
public:
    int minAddToMakeValid(string S)
    {
        vector<int> stack;

        for (int i = 0; i < S.size(); i++)
        {
            if (!stack.empty() && stack.back() == '(' && S[i] == ')')
            {
                stack.pop_back();
            }
            else
            {
                stack.push_back(S[i]);
            }
        }

        return stack.size();
    }
};

File: Leetcode/921. Minimum Add to Make Parentheses Valid/solution.go
func minAddToMakeValid(S string) int {
    stack := make([]int, 0)
    
    for _, char := range S {
        if len(stack) > 0 && stack[len(stack) - 1] == '(' && char == ')' {
            stack = stack[:len(stack) - 1]
        } else {
            stack = append(stack, int(char))
        }
    }
    
    return len(stack)
}